src/nvim/path.c

```diff
@@ -1267,13 +1250,12 @@ int gen_expand_wildcards(int num_pat, char **pat, int *num_file, char ***file, i
         if (p == NULL) {
           p = (char_u *)pat[i];
         }
+      } else {
```

Comment:
Oh, this `else` is after the `if (p == NULL)`, so:
```suggestion
        }
      } else {
```
And the indent and braces below also need change.